src/Groups/GroupsList/GroupsList.js

```diff
@@ -40,92 +57,130 @@ const columnMapping = {
   code: <FormattedMessage id="ui-finance.groups.list.code" />,
 };
 
-const renderNavigation = () => (
-  <FinanceNavigation />
-);
+class GroupsList extends Component {
+  static propTypes = {
+    stripes: PropTypes.object,
+    intl: intlShape.isRequired,
+    history: ReactRouterPropTypes.history.isRequired,
+    mutator: PropTypes.object.isRequired,
+    resources: PropTypes.object.isRequired,
+  };
+
+  static manifest = Object.freeze({
+    query: {
+      initialValue: {
+        query: '',
+        filters: '',
+        sort: '',
+      },
+    },
+    resultCount: { initialValue: INITIAL_RESULT_COUNT },
+    records: {
+      type: 'okapi',
+      clear: true,
+      records: 'groups',
+      recordsRequired: '%{resultCount}',
+      path: GROUPS_API,
+      perRequest: RESULT_COUNT_INCREMENT,
+      throwErrors: false,
+      GET: {
+        params: {
+          query: makeQueryFunction(
+            'cql.allRecords=1',
+            groupsSearchTemplate,
+            {},
+            filterConfig,
+          ),
+        },
+        staticFallback: { params: {} },
+      },
+    },
+  });
 
-const GroupsList = ({
-  resources,
-  mutator,
-  stripes,
-  history,
-}) => {
-  const showToast = useShowToast();
+  constructor(props) {
+    super(props);
 
-  const onCreate = useCallback(
-    async (group) => {
-      try {
-        const savedGroup = await mutator.records.POST(group);
+    this.getActiveFilters = getActiveFilters.bind(this);
+    this.handleFilterChange = handleFilterChange.bind(this);
+    this.changeSearchIndex = changeSearchIndex.bind(this);
+    this.callout = React.createRef();
+    this.showToast = showToast.bind(this);
```

Comment:
don't we added this to index and use a common hook?

Reply:
But we use hooks for functional components

Reply:
yes, you right, let's keep it the olds style till we switch on functional component list